instructions/rate-limiting.adoc

```diff
@@ -1,98 +1,203 @@
-
-~~~yaml
-- dimensions:
-    destination: recommendation
-    destinationVersion: v2
-    source: preference
-  maxAmount: 1
-  validDuration: 1s
-validDuration: 1s
-~~~
-
-You can also conditionally rate limit based on other dimensions, such as:
-
-* Source and Destination project names (e.g. to limit developer projects from overloading the production services during testing)
-* Login names (e.g. to limit certain customers or classes of customers)
-* Source/Destination hostnames, IP addresses, DNS domains, HTTP Request header values, protocols
-* API paths
-* [Several other attributes](https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html){:target="_blank"}
-
-## Attempt to access app
-
-With the quota in place for `recommendation:v2` let's hit it as fast as we can and observe the `429 Too Many Requests` HTTP
-error generated as a result of the quota:
-
-~~~bash
-for i in $(seq 20); do
-  curl "http://customer-${ISTIO_LAB_PROJECT}.{{APPS_SUFFIX}}"
-done
-~~~
-
-The first or second attempt to access `recommendation:v2` will succeed:
-
-`customer => preference => recommendation v2 from '5445bf797-zmvdq': 635`
-
-But the next one (which occurs less than 1 second later) will fail:
-
-`customer => 503 preference => 429 RESOURCE_EXHAUSTED:Quota is exhausted for: requestcount`
-
-This is because our quota only allows 1 request per second to succceed. After 1 second elapses,
-the next call to `v2` will succeed but then start failing again. So you will only get one successful
-call to `v2` every second. Rate limiting in action!
-
-## Cleanup
-
-Remove the rate limit:
-
-~~~bash
-oc delete -f ${ISTIO_LAB_HOME}/src/istiofiles/rate_limit_rule.yml
-oc delete -f ${ISTIO_LAB_HOME}/src/istiofiles/recommendation_rate_limit_handler.yml
-~~~
-
-# References
-
-* [Red Hat OpenShift](https://openshift.com){:target="_blank"}
-* [Learn Istio on OpenShift](https://learn.openshift.com/servicemesh){:target="_blank"}
-* [Istio Homepage](https://istio.io){:target="_blank"}
+= Rate-Limiting/Policy
+include::_attributes.adoc[]
+
+Apply request rate limits to a service
+
+:toc:
+
+== What we will learn in this module
+This module will provide instruction on how to limit the amount of traffic to a particular service in
+terms of requests per time period.
+
+[IMPORTANT]
+.Before Start
+====
+You should have only the following virtualservices and destinationrules in the `istio-tutorial` namespace
+
+[source, bash,subs="+macros,+attributes"]
+----
+$ oc -n istio-tutorial get destinationrule
+No resources found.
+
+$ oc -n istio-tutorial get virtualservice
+NAME       GATEWAYS             HOSTS   AGE
+customer   [customer-gateway]   [*]     18h
+----
+====
+
+[#ratelimiting]
+== Rate Limiting
+
+IMPORTANT: The Rate Limiting rules take some time to be applied and reflected. Be patient here!
+
+Here we will limit the number of concurrent requests into recommendation v2
+
+Now apply the rate limiting. To see how this is done,
```

Comment:
nit: Style is different between the way info is presented here and fault-injection